pkg/handlers/v1/subnet.go

```diff
@@ -0,0 +1,77 @@
+package v1
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+type subnetConfiguration struct {
+	CIDRBlock string `json:"cidrBlock"`
+}
+
+type subnetConfigurationDiff struct {
+	PreviousValue *subnetConfiguration `json:"previousValue"`
+	UpdatedValue  *subnetConfiguration `json:"updatedValue"`
+	ChangeType    string               `json:"changeType"`
+}
+
+type subnetTransformer struct{}
+
+func (t subnetTransformer) Create(event awsConfigEvent) (Output, bool, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, false, err
+	}
+
+	var config subnetConfiguration
+	if err := json.Unmarshal(event.ConfigurationItem.Configuration, &config); err != nil {
+		return Output{}, false, err
+	}
+
+	change := extractSubnetInfo(&config)
+	change.ChangeType = added
+	output.Changes = append(output.Changes, change)
+	return output, false, nil
+}
+
+func (t subnetTransformer) Update(event awsConfigEvent) (Output, bool, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, false, err
+	}
+
+	return output, false, nil
+}
+
+func (t subnetTransformer) Delete(event awsConfigEvent) (Output, bool, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, false, err
+	}
+
+	changeProps := event.ConfigurationItemDiff.ChangedProperties
+
+	configDiffRaw, ok := changeProps["Configuration"]
+	if !ok {
+		return Output{}, false, errors.New("invalid configuration diff")
+	}
+
+	var configDiff subnetConfigurationDiff
+
+	if err := json.Unmarshal(configDiffRaw, &configDiff); err != nil {
+		return Output{}, false, err
+	}
+
+	change := extractSubnetInfo(configDiff.PreviousValue)
+	change.ChangeType = deleted
+	output.Changes = append(output.Changes, change)
+	fmt.Println(output)
+	return output, false, nil
+}
+
+func extractSubnetInfo(config *subnetConfiguration) Change {
+	return Change{
```

Comment:
I think this needs to pull out the relationship info at this step as well. The ENI handler does something similar https://github.com/asecurityteam/awsconfig-transformerd/blob/master/pkg/handlers/v1/eni.go#L154

Trying this out locally at least I'm not seeing related resources show up with the test data. I'm dashing this off before a meeting, sorry if I missed anything!!!

Reply:
Updated! Thanks!